app/src/main/java/mozilla/lockbox/autofill/FillResponseBuilder.kt

```diff
@@ -31,24 +41,35 @@ open class FillResponseBuilder(
         presentation.setTextViewText(R.id.autofill_cta, authenticationText)
 
         val sender = IntentBuilder.getAuthIntentSender(context, this)
-        responseBuilder.setAuthentication(autofillIds(), sender, presentation)
+
+        responseBuilder.setAuthentication(parsedStructure.autofillIds, sender, presentation)
+        if (enableSave) {
+            responseBuilder.setSaveInfo(buildSaveInfo())
+        }
```

Comment:
This is duplicated here (`fun buildAuthenticationFillResponse`) and below in `fun buildFallbackFillResponse` and `fun buildFilteredFillResponse`. Consider making a single check on creation/initialization before these are even called? Alternatively, break this out into its own method.